### PR TITLE
enable login via irc PASS command during handshake instead of PRIVMSG

### PR DESCRIPTION
**Pull request by dgruss**
this allows to just pass credentials as the server password in the irc client

### CODE REVIEW

**Comment by dgruss** [comment]
moved to this project, from this one https://github.com/42wim/matterircd/pull/171

File: server.go [hunk 1]
```diff
@@ -446,7 +448,16 @@ func (s *server) handshake(u *User) error {
...
 		s.u = u
-		return s.welcome(u)
+
+		err = s.welcome(u)
+		if err == nil && u.Pass != nil {
+			service := "mattermost"
+			if len(u.Pass) == 1 {
+				service = "slack"
+			}
+			login(u, u, u.Pass, service)
```
Comment (42wim) [inline server.go]: I would change this to
login(u, &User{Nick: service, User: service, Real: service, Host: "service", channels: map[Channel]struct{}{}}, u.Pass, service)
The login OK message needs to come from the service, not the user

**Comment by 42wim** [comment]
Looks good, thanks for your work!

**Comment by keeperofdakeys** [comment]
@dgruss I tried using this feature from irssi, but I couldn't get it to work. Could you tell me what syntax I need to get it to work?

Once I've got it working, I'll send a PR to document it in the README.

**Comment by dgruss** [comment]
On 11.05.2018 02:36, keeperofdakeys wrote:
> @dgruss I tried using this feature from irssi, but I couldn't get it to work. Could you tell me what syntax I need to get it to work?

Same syntax as the arguments of the login command.

**Pull request merged**